File: Python/Person.py
class Person:
    # Class variable
    population = 0

    def __init__(self, name, age):
        # Instance variables
        self.name = name
        self.age = age
        Person.population += 1

    # Instance method to get the person's name
    def get_name(self):
        return self.name

    # Instance method to set the person's name
    def set_name(self, name):
        self.name = name

    # Instance method to get the person's age
    def get_age(self):
        return self.age

    # Instance method to set the person's age
    def set_age(self, age):
        if age > 0:
            self.age = age
        else:
            print("Age must be positive.")

    # Instance method to display the person's details
    def display_details(self):
        print(f"Name: {self.name}")
        print(f"Age: {self.age}")

    # Class method to get the current population
    @classmethod
    def get_population(cls):
        return cls.population


# Main function to demonstrate the usage of the Person class
def main():
    # Create two Person objects
    person1 = Person("Alice", 30)
    person2 = Person("Bob", 25)

    # Display the details of person1
    person1.display_details()

    # Display the details of person2
    person2.display_details()

    # Change the age of person2
    person2.set_age(35)

    # Display the updated details of person2
    person2.display_details()

    # Display the current population
    print(f"Current population: {Person.get_population()}")

if __name__ == "__main__":
    main()

File: Java/Search/Person.java
// Define the Person class
public class Person {
    // Instance variables
    private String name;
    private int age;

    // Constructor to initialize the instance variables
    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    // Instance method to get the person's name
    public String getName() {
        return name;
    }

    // Instance method to set the person's name
    public void setName(String name) {
        this.name = name;
    }

    // Instance method to get the person's age
    public int getAge() {
        return age;
    }

    // Instance method to set the person's age
    public void setAge(int age) {
        if (age > 0) {
            this.age = age;
        } else {
            System.out.println("Age must be positive.");
        }
    }

    // Instance method to display the person's details
    public void displayDetails() {
        System.out.println("Name: " + name);
        System.out.println("Age: " + age);
    }

    // Static method to compare the ages of two persons
    public static Person olderPerson(Person p1, Person p2) {
        if (p1.getAge() > p2.getAge()) {
            return p1;
        } else {
            return p2;
        }
    }

    // Main method to demonstrate the usage of the Person class
    public static void main(String[] args) {
        // Create two Person objects using the constructor
        Person person1 = new Person("Alice", 30);
        Person person2 = new Person("Bob", 25);

        // Display the details of person1
        person1.displayDetails();

        // Display the details of person2
        person2.displayDetails();

        // Change the age of person2
        person2.setAge(35);

        // Display the updated details of person2
        person2.displayDetails();

        // Compare the ages of person1 and person2
        Person older = Person.olderPerson(person1, person2);
        System.out.println("The older person is: " + older.getName());
    }
}
